File: 07/main.hs
import Data.List (sort)
import Data.List.Split (splitOn)
import Data.IntMap (IntMap, fromList, (!), insert)

(|>) :: (a -> b) -> (b -> c) -> a -> c
f |> g = g . f

f :: [Int] -> Int
f xs =  sum $ map ((median -) |> abs) xs
    where 
        median = xs !! (length xs `div` 2)

g :: [Int] -> Int
g xs = minimum ys
    where
        ys = map (dSum |> sum) xs
        dSum x = map ((x -) |> abs |> sumN) xs
        sumN n = (n*(n+1)) `div` 2

main :: IO ()
main = interact $
    splitOn "," |> map read |> sort |> (\xs -> [f xs, g xs]) |> map show |> unlines

File: 07/main.py
#!/usr/bin/env python3
with open('data.txt') as f:
    xs = [int(x) for x in f.read().split(',')]

xs.sort()
print(sum(abs(x - xs[len(xs)//2]) for x in xs))
print(min(sum(abs((x-xs[i])*(abs(x-xs[i])+1))//2 for x in xs) for i in range(len(xs))))